Websites/webkit.org/wp-content/themes/webkit/standards-positions.php

```diff
@@ -0,0 +1,176 @@
+  const table = addElement(fragment, "table");
+  const headerRow = addElement(addElement(table, "thead"), "tr");
+  for (const heading of ["Issue", "Specification", "Concerns", "Position", "Topics", "Venues", "Self-link"]) {
+    addElement(headerRow, "th", { scope: "col" }, heading);
+  }
+  addElement(table, "tbody");
+  return fragment;
+}
+
+function createFilterTextRE(filterText) {
+  if (filterText) {
+    try {
+      return new RegExp(filterText, "i");
+    } catch(e) {}
+  }
+  return null; 
+}
+
+function createStandardsPositionsTableRows(issues, filterText = "", filterTopic = "") {
+  const fragment = new DocumentFragment();
+  const filterTextRE = createFilterTextRE(filterText);
+  for (const issue of issues) {
+    if (filterTextRE && issue["title"].search(filterTextRE) === -1) {
+      continue;
+    }
+    if (filterTopic && !issue["topics"].includes(filterTopic)) {
+      continue;
+    }
+    const id = issue["id"].substring("https://github.com/WebKit/standards-positions/issues/".length);
+    
+    const row = addElement(fragment, "tr", { id: `position-${id}` });
+    addElement(row, "td", {}, createElement("a", { href:issue["id"] }, `#${id}`));
+    if (issue["url"] !== null) {
+      addElement(row, "td", {}, createElement("a", { href:issue["url"] }, issue["title"]));
+    } else {
+      addElement(row, "td", {}, issue["title"]);
+    }
+    addElement(row, "td", {}, issue["concerns"].join(", "));
+    if (issue["position"] !== null) {
+      addElement(row, "td", { class: issue["position"].replace(" ", "-") }, createElement("span", {}, issue["position"]));  
+    } else {
+      addElement(row, "td");      
+    }
+    addElement(row, "td", {}, issue["topics"].join(", "));
+    addElement(row, "td", {}, issue["venues"].join(", "));
+    addElement(row, "td", {}, createElement("a", { href: `#position-${id}`, title: `Link to position: ${issue["title"]}` }, "⚓︎"));
+  }
+  return fragment;
+}
+
+let standardsPositions = null;
+let tbody = null;
+const container = document.querySelector("main"); 
+const noScriptChild = container.querySelector("p");
+const loadingChild = createElement("p", {}, "Please hold while we fetch data from GitHub.");
+noScriptChild.replaceWith(loadingChild);
+
+function updateStandardsPositionsTableRows() {
+  const filterText = document.getElementById("filter").value;
+  const filterTopic = document.getElementById("filter-topic").value;
+  tbody.replaceChildren(createStandardsPositionsTableRows(standardsPositions, filterText, filterTopic));
+}
+
+function updateTopicFilter() {
+  const topics = [];
+  for (const issue of standardsPositions) {
+    for (const topic of issue["topics"]) {
+      if (!topics.includes(topic)) {
+        topics.push(topic);
+      }
+    }
+  }
+  topics.sort();
+  const select = document.getElementById("filter-topic");
+  addElement(select, "option", { value:"" }, "all");
+  for (const topic of topics) {
+    addElement(select, "option", {}, topic);
+  }
+}
```

Comment:
Bit shorter: 

```suggestion
function updateTopicFilter() {
  const topics = new Set([standardsPositions.map(({topics}) => topics).flat()]);
  const select = document.getElementById("filter-topic");
  addElement(select, "option", { value:"" }, "all");
  for (const topic of topics.sort()) {
    addElement(select, "option", {}, topic);
  }
}
```

Reply:
I find this a lot harder to parse, personally.

Reply:
Hmm, interesting. 

I might just be used to doing things declaratively: with the proposed change one can see declared exactly what's going to happen: `map().flat().sort()` into a unique `Set`. In the current code, everything is all implied, which means actually having to read through it to figure out what it's doing.

I know some folks don't like the "functional programming" style, so could be a matter of preference.

Reply:
I think my main problem is that I have a hard time reading the `map()` function and its argument. Whereas a nested for-of loop is very straightforward to read. I'm happy to change it though, not a hill worth dying on.

Reply:
Right, yeah. A lot of people dislike `.map()` and friends for that reason. Would this help make it clear?:

```JS
const toTopics = ({topics}) => topics;
const topics = new Set([standardsPositions.map(toTopics).flat().sort()]);
```